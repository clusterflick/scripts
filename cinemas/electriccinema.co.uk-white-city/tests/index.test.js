/** @jest-environment setup-polly-jest/jest-environment-node */
const { setupPolly, schemaValidate } = require("../../../common/test-utils");
const { sortAndFilterMovies } = require("../../../common/utils");
const { retrieve, transform, attributes } = require("..");

const isRecording = false;

describe(attributes.name, () => {
  setupPolly(isRecording, __dirname);
  jest.useFakeTimers().setSystemTime(new Date("2025-01-23"));

  it(
    "retrieve and transform",
    async () => {
      const moviePages = await retrieve();

      // Make sure the input looks roughly correct
      expect(moviePages).toBeTruthy();
      expect(Object.keys(moviePages.filmData.films)).toHaveLength(7);

      const output = sortAndFilterMovies(await transform(moviePages, {}));
      const data = JSON.parse(JSON.stringify(output));

      // Make sure the data looks roughly correct
      expect(data).toHaveLength(6); // One of the entries has no performances

      expect(schemaValidate(data)).toBe(true);
      expect(data).toMatchSnapshot();
    },
    isRecording ? 120_000 : undefined,
  );
});
