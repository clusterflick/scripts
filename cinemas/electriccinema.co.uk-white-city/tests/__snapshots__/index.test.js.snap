// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Electric Cinema White City retrieve and transform 1`] = `
[
  {
    "overview": {
      "actors": [],
      "categories": [],
      "classification": "15",
      "directors": [
        "James Mangold",
      ],
      "year": "2024",
    },
    "performances": [
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/film/a-complete-unknown/",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": true,
        },
        "time": 1737653400000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/film/a-complete-unknown/",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": true,
        },
        "time": 1737665400000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771117",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1737740700000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771118",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1737752400000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771123",
        "notes": "",
        "screen": "3",
        "status": {
          "soldOut": false,
        },
        "time": 1737818100000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771121",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1737823500000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771122",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1737837000000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771125",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1737899100000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771126",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1737911700000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771127",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1737923400000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771147",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1737999900000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771148",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1738011600000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771171",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1738085400000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771172",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1738097100000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771176",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1738172700000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771177",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1738184400000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771182",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1738270800000,
      },
    ],
    "title": "A Complete Unknown",
    "url": "https://www.electriccinema.co.uk/film/a-complete-unknown/",
  },
  {
    "overview": {
      "actors": [],
      "categories": [],
      "classification": "15",
      "directors": [
        "Jesse Eisenberg",
      ],
      "year": "2024",
    },
    "performances": [
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/769979",
        "notes": "",
        "screen": "3",
        "status": {
          "soldOut": false,
        },
        "time": 1737655200000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771119",
        "notes": "",
        "screen": "3",
        "status": {
          "soldOut": false,
        },
        "time": 1737742500000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771120",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1737814500000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771124",
        "notes": "",
        "screen": "3",
        "status": {
          "soldOut": false,
        },
        "time": 1737829800000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771128",
        "notes": "",
        "screen": "3",
        "status": {
          "soldOut": false,
        },
        "time": 1737892800000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771129",
        "notes": "",
        "screen": "3",
        "status": {
          "soldOut": false,
        },
        "time": 1737912600000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771149",
        "notes": "",
        "screen": "2",
        "status": {
          "soldOut": false,
        },
        "time": 1737997200000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771173",
        "notes": "",
        "screen": "3",
        "status": {
          "soldOut": false,
        },
        "time": 1738089000000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771178",
        "notes": "",
        "screen": "3",
        "status": {
          "soldOut": false,
        },
        "time": 1738174500000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771180",
        "notes": "",
        "screen": "2",
        "status": {
          "soldOut": false,
        },
        "time": 1738256700000,
      },
    ],
    "title": "A Real Pain",
    "url": "https://www.electriccinema.co.uk/film/a-real-pain/",
  },
  {
    "overview": {
      "actors": [],
      "categories": [],
      "classification": "18",
      "directors": [],
      "year": "2024",
    },
    "performances": [
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/film/babygirl/",
        "notes": "",
        "screen": "3",
        "status": {
          "soldOut": true,
        },
        "time": 1737664200000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771165",
        "notes": "",
        "screen": "3",
        "status": {
          "soldOut": false,
        },
        "time": 1737751500000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771167",
        "notes": "",
        "screen": "3",
        "status": {
          "soldOut": false,
        },
        "time": 1737808200000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771166",
        "notes": "",
        "screen": "3",
        "status": {
          "soldOut": false,
        },
        "time": 1737838800000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771168",
        "notes": "",
        "screen": "3",
        "status": {
          "soldOut": false,
        },
        "time": 1737901800000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771169",
        "notes": "",
        "screen": "3",
        "status": {
          "soldOut": false,
        },
        "time": 1737921600000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771170",
        "notes": "",
        "screen": "3",
        "status": {
          "soldOut": false,
        },
        "time": 1738010700000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771174",
        "notes": "",
        "screen": "3",
        "status": {
          "soldOut": false,
        },
        "time": 1738098600000,
      },
      {
        "accessibility": {
          "babyFriendly": true,
        },
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771175",
        "notes": "Electric Scream!",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1738153800000,
      },
      {
        "accessibility": {
          "subtitled": true,
        },
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771207",
        "notes": "Electric Subtitled",
        "screen": "2",
        "status": {
          "soldOut": false,
        },
        "time": 1738159200000,
      },
      {
        "accessibility": {
          "subtitled": true,
        },
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771179",
        "notes": "Electric Subtitled",
        "screen": "3",
        "status": {
          "soldOut": false,
        },
        "time": 1738183500000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771181",
        "notes": "",
        "screen": "3",
        "status": {
          "soldOut": false,
        },
        "time": 1738269900000,
      },
    ],
    "title": "Babygirl",
    "url": "https://www.electriccinema.co.uk/film/babygirl/",
  },
  {
    "overview": {
      "actors": [],
      "categories": [],
      "directors": [
        "Michael Morris",
      ],
      "year": "2025",
    },
    "performances": [
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/film/bridget-jones-mad-about-the-boy/",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1739469600000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/film/bridget-jones-mad-about-the-boy/",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1739480400000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/film/bridget-jones-mad-about-the-boy/",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1739556000000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/film/bridget-jones-mad-about-the-boy/",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1739567700000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/film/bridget-jones-mad-about-the-boy/",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1739633400000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/film/bridget-jones-mad-about-the-boy/",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1739644200000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/film/bridget-jones-mad-about-the-boy/",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1739655000000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/film/bridget-jones-mad-about-the-boy/",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1739715600000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/film/bridget-jones-mad-about-the-boy/",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1739727000000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/film/bridget-jones-mad-about-the-boy/",
        "notes": "",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1739737800000,
      },
    ],
    "title": "Bridget Jones: Mad About the Boy",
    "url": "https://www.electriccinema.co.uk/film/bridget-jones-mad-about-the-boy/",
  },
  {
    "overview": {
      "actors": [],
      "categories": [],
      "classification": "U",
      "directors": [
        "James Bobin",
      ],
      "year": "2014",
    },
    "performances": [
      {
        "accessibility": {
          "babyFriendly": true,
        },
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/769990",
        "notes": "Electric Kids Club",
        "screen": "1",
        "status": {
          "soldOut": false,
        },
        "time": 1737889200000,
      },
    ],
    "title": "Muppets Most Wanted",
    "url": "https://www.electriccinema.co.uk/film/muppets-most-wanted/",
  },
  {
    "overview": {
      "actors": [],
      "categories": [],
      "classification": "15",
      "directors": [
        "Robert Eggers",
      ],
      "year": "2024",
    },
    "performances": [
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771712",
        "notes": "",
        "screen": "3",
        "status": {
          "soldOut": false,
        },
        "time": 1737999000000,
      },
      {
        "accessibility": {},
        "bookingUrl": "https://www.electriccinema.co.uk/tickets/771713",
        "notes": "",
        "screen": "3",
        "status": {
          "soldOut": false,
        },
        "time": 1738258200000,
      },
    ],
    "title": "Nosferatu",
    "url": "https://www.electriccinema.co.uk/film/nosferatu/",
  },
]
`;
